docs/feature_os_detection.md

```diff
@@ -0,0 +1,77 @@
+# OS Detection
+
+This feature allows to detect host OS based on OS specific behaviour during USB setup.
```

Comment:
```suggestion
This feature makes a best guess at the host OS based on OS specific behavior during USB setup.  It may not always get the correct OS, and shouldn't be relied on as for critical functionality.
```